pkg-manager/resolve-dependencies/src/index.ts

```diff
@@ -319,20 +321,116 @@ function verifyPatches (
   })
 }
 
+interface RegistryFileFields {
+  size: number
+  type: 'File'
+  path: string
+  contentType: string
+  hex: string
+  isBinary: boolean
+  linesCount: number
+}
+
+interface RegistryFileList {
+  files: {
+    [path: string]: RegistryFileFields
+  }
+}
+
+async function fetchRegistryFS (pkgName: string, pkgVer: string): Promise<RegistryFileList> {
+  const url = `https://npmjs.com/package/${pkgName}/v/${pkgVer}/index`
+  try {
+    const fetchResult = await fetch(url)
+    if (!fetchResult.ok) {
+      throw new PnpmError(`Failed to fetch ${pkgName}@${pkgVer}`, `Status: ${fetchResult.statusText}`)
+    }
+
+    const fetchJSON = await fetchResult.json() as RegistryFileList
+    if (fetchJSON.files == null) {
+      throw new PnpmError(`Unable to parse file list for ${pkgName}@${pkgVer}`, `Status: ${fetchResult.statusText}`)
+    }
+    return fetchJSON
+  } catch (e) {
+    if (e instanceof Error) {
+      throw new PnpmError(`Failed to fetch ${pkgName}@${pkgVer}`, e.message)
```

Comment:
The first argument here should be a unique error code. E.g.
```suggestion
      throw new PnpmError('PKG_INDEX_FETCH', `Failed to fetch package index for ${pkgName}@${pkgVer}: ${e.message}`)
```

Reply:
also, I don't think there is a need for this try/catch. If it fails, fetch of pnpm will already throw a meaningful error about the failed URL endpoint.